skbeam/core/accumulators/binned_statistic.py

```diff
@@ -130,7 +140,12 @@ def __init__(self, sample, statistic='mean',
         # Compute the bin number each sample falls into.
         Ncount = {}
         for i in np.arange(self.D):
-            Ncount[i] = np.digitize(sample[:, i], self.edges[i])
+            # Apply mask in a non-ideal way by setting value outside range.
+            # Would be better to do this using bincount "weights", perhaps.
+            thissample = sample[:, i]
+            thissample[mask == 0] = (self.edges[i][0] -
```

Comment:
suggest putting the `mask` test here.  If there is no mask you will do a whole bunch of work to not do any work.  `if mask is not None` is cheaper